src/lib.rs

```diff
@@ -238,6 +238,82 @@ impl<T: Clone + Integer + PrimInt> Ratio<T> {
     }
 }
 
+macro_rules! pow_impl {
+    ($exp: ty) => {
+        pow_impl!($exp, $exp);
+    };
+    ($exp: ty, $unsigned: ty) => {
+        impl<T: Clone + Integer + Pow<$unsigned, Output = T>> Pow<$exp> for Ratio<T> {
+            type Output = Ratio<T>;
+            #[inline]
+            fn pow(self, expon: $exp) -> Ratio<T> {
+                match expon.cmp(&0) {
+                    cmp::Ordering::Equal => One::one(),
+                    cmp::Ordering::Less => {
+                        let expon = expon.wrapping_abs() as $unsigned;
+                        Ratio::new_raw(
+                            Pow::pow(self.denom, expon),
+                            Pow::pow(self.numer, expon),
+                        )
+                    },
+                    cmp::Ordering::Greater => {
+                        Ratio::new_raw(
+                            Pow::pow(self.numer, expon as $unsigned),
+                            Pow::pow(self.denom, expon as $unsigned),
+                        )
+                    }
+                }
+            }
+        }
+        impl<'a, T: Clone + Integer + Pow<$unsigned, Output = T>> Pow<$exp> for &'a Ratio<T> {
+            type Output = Ratio<T>;
+            #[inline]
+            fn pow(self, expon: $exp) -> Ratio<T> {
+                Pow::pow(self.clone(), expon)
+            }
+        }
+        impl<'a, T: Clone + Integer + Pow<$unsigned, Output = T>> Pow<&'a $exp> for Ratio<T> {
```

Comment:
This does not need `Clone`, does it?

Reply:
Everything requires `Clone`, even `Ratio::new`.  I'd prefer to keep it for now.  We could someday relax this without it being a breaking change, but not the other way.